### PR TITLE
Add a config to allow additional parameters in OAuth2 error url

### PR DESCRIPTION
**Pull request by Ashi1993**
### Proposed changes in this pull request

- Add a config to allow additional parameters in OAuth2 error url
related to https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1379

Fixes wso2/product-is#8326 and wso2-enterprise/financial-open-banking#5289

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [x] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [x] **Is the PR labeled correctly?**

#### Functionality

- [x] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [x] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 1]
```diff
@@ -126,6 +126,8 @@
 
   "oauth.dcrm.application_role_permission_required_to_view": true,
 
+  "oauth.allow_additional_params_from_error_url": false,
```
Comment (janakamarasena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:129]: we can remove the default value as this is properly handled in the code. adding the default value will always add it to the identity.xml file.

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 2]
```diff
@@ -552,6 +552,11 @@
         <DCRM>
             <ApplicationRolePermissionRequiredToView>{{oauth.dcrm.application_role_permission_required_to_view}}</ApplicationRolePermissionRequiredToView>
         </DCRM>
+
+        <!--
+        By enabling this config, additional parameters will be append to OAuth2 error url
+        -->
+        <AllowAdditionalParamsFromErrorUrl>false</AllowAdditionalParamsFromErrorUrl>
```
Comment (janakamarasena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:559]: We can comment this config as from the code level we are properly handling the default

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 3]
```diff
@@ -552,6 +552,11 @@
         <DCRM>
             <ApplicationRolePermissionRequiredToView>{{oauth.dcrm.application_role_permission_required_to_view}}</ApplicationRolePermissionRequiredToView>
         </DCRM>
+
+        <!--
+        By enabling this config, additional parameters will be append to OAuth2 error url
```
Comment (janakamarasena) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:557]: lets mentioned the default value and the current params that will be added if the config is set to true

Comment (Ashi1993) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:557]: done

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml (hunk 2 above)
Comment (Ashi1993) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:559]: done

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json (hunk 1 above)
Comment (Ashi1993) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:129]: done

**Pull request merged**